src/platform/pret_sdl/sdl2.c

```diff
@@ -593,9 +593,9 @@ void DmaSet(int dmaNum, const void *src, void *dest, u32 control)
         return;
     }
 
-    (&REG_DMA0SAD)[dmaNum * 3] = (vu32)src;
-    (&REG_DMA0DAD)[dmaNum * 3] = (vu32)dest;
-    (&REG_DMA0CNT)[dmaNum * 3] = (vu32)control;
+    (&REG_DMA0SAD)[dmaNum * 3] = *(vu32 *)&src;
+    (&REG_DMA0DAD)[dmaNum * 3] = *(vu32 *)&dest;
+    (&REG_DMA0CNT)[dmaNum * 3] = *(vu32 *)&control;
```

Comment:
Wouldn't (volatile uintptr_t) do the trick as well, instead of casting back-and-forth between the pointers?

Reply:
Yep you are right!